### ISSUE TITLE
2.x Composer.json vendor install path

### ISSUE THREAD

**Issue by xhs345** [cakephp/cakephp#8765]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 2.8.3
* Platform and Target: Windows 7

This issue is related to: https://github.com/cakephp/cakephp/pull/5765

### What you did
After downloading a fresh install of CakePHP 2.8.3 I ran 'composer install' to install all dependencies. After that I tried running PHPUnit through the shell (Console\cake test core), which failed with multiple errors (see below). I fixed the error by manually moving the installed files from the folder 'vendor' to the folder 'vendors'

### Expected Behavior
No error when running the tests

### Actual Behavior
Error message:
```
Warning Error: include(PHPUnit\Autoload.php): failed to open stream: No such file or directory in [C:\Users\aslotty\Code\cakephp\lib\Cake\TestSuite\CakeTestSuiteDispatcher.php, line 162]

Warning Error: include(): Failed opening 'PHPUnit\Autoload.php' for inclusion (include_path='C:\Users\aslotty\Code\cakephp\lib;.;C:\xampp\php\PEAR') in [C:\Users\aslotty\Code\cakephp\lib\Cake\TestSuite\CakeTestSuiteDispatcher.php, line 162]

Error: Please install PHPUnit framework v3.7 (http://www.phpunit.de)
#0 C:\Users\aslotty\Code\cakephp\lib\Cake\Console\ShellDispatcher.php(211): TestShell->initialize()
#1 C:\Users\aslotty\Code\cakephp\lib\Cake\Console\ShellDispatcher.php(66): ShellDispatcher->dispatch()
#2 C:\Users\aslotty\Code\cakephp\app\Console\cake.php(47): ShellDispatcher::run(Array)
#3 {main}
```

### Suggestion
I don't know all the implications on how to fix this, but one way could be to set 
```
  "extra": {
    "installer-paths": {
```
in the composer.json

**Comment by davidsteinsland** [comment]
My bet is one of:

1. You did not require `vendor/autoload.php`
2. You did not install using `--dev`

**Comment by xhs345** [comment]
1. Yes, the autoload.php is not being loaded by Cake, but I expect that this would be the default behavior after using composer. I didn't see any indication in the manual that I have to do additional configuration
2. Not required, `--dev` is default behavior when running `composer install`

**Comment by markstory** [comment]
The [composer installation instructions](http://book.cakephp.org/2.0/en/installation/advanced-installation.html#installing-cakephp-with-composer) cover the required steps which include 

```php
require APP . 'Vendor/autoload.php';
```

**Comment by xhs345** [comment]
Now I see. I downloaded the release and then just ran the included `composer.json` to install all dependencies. I skipped that part you linked, because it talks about installing cakephp in the beginning so I thought it would not be relevant to me.

> The composer installation instructions cover the required steps which include 
> require APP . 'Vendor/autoload.php';

This wouldn't have helped, because the install path of PHPUnit with the release-included `composer.json` still is `ROOT . Vendor/phpunit`
